packages/taquito-michelson-encoder/src/tokens/comparable/timestamp.ts

```diff
@@ -12,7 +12,10 @@ export class TimestampToken extends ComparableToken {
   }
 
   public Execute(val: { string?: string; int?: string }) {
-    if (val.string) {
+    if (val.string && /^\d{10}(\.\d)?$/.test(val.string)) {
+      const timestamp = +val.string;
+      return new Date(Number(timestamp) * 1000).toISOString();
```

Comment:
"+" and "Number" have the same behavior of converting the string to a number. I suggest keeping only one of them.

Reply:
you are correct, I missed the `Number()` casting